### DISCUSSION TITLE
Is there any possibility that one individual receives the same chunk from different users?

### DISCUSSION THREAD

**Discussion by WY0412** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#27] [Q&A]
For example, peer A initially selected to download chunk 1 from peer B, could A send request to peer C for another copy of chunk 1 because of the high latency between A and B? And another question, is the RTT evaluated based on one single user or the whole network? ie, should we evaluate the RTT as global RTT shared among all peers or each peer has its own RTT ?

**Comment by TANSixu** [comment]
Hi, thanks for reaching out!

For the first question, you can use any reasonable scheme as long as you can successfully download the chunk. However, be advised that you do not need to focus specifically on peer selection. Note that:
1. Comprehensive tests only check if you can finish all downloading tasks, speed does not matter.
2. In bonus test, we only care about congestion control, and a better peer selection will not be helpful.

There is exactly a scenario you **NEED** to change peer selection: **Peer crash**. In your example, if peer B crashes in the downloading process, peer A should be able to sense the crash and start to download from peer C.

For the second question, RTT is **per connnection**. In your example, A-B and A-C should have different RTT estimations. You should use the RTT value to estimate a reasonable timeout for this connection. Note that if timeout is set in the argument when starting a peer, you should use the pre-set value.

Please continue this discussion thread if you have further questions regarding this.
